Sources/Sentry/SentryScope.m

```diff
@@ -319,6 +326,33 @@ - (void)addAttachment:(SentryAttachment *)attachment
     }
 }
 
+- (void)setTransaction:(id<SentrySpan>)transaction
+{
+    self.currentTransaction = transaction;
+    [self notifyListeners];
+}
+
+- (void)setTransactionName:(NSString *)transactionName
+{
+    if (self.currentTransaction != nil &&
+        [self.currentTransaction isKindOfClass:[SentryTracer class]]) {
+        [(SentryTracer *)self.currentTransaction setName:transactionName];
+    }
+    self.fallbackTransactionName = transactionName;
+    [self notifyListeners];
+}
+
+- (nullable id<SentrySpan>)span
+{
+    if (self.currentTransaction != nil &&
+        [self.currentTransaction isKindOfClass:[SentryTracer class]]) {
+        id latestSpan = [(SentryTracer *)self.currentTransaction getLatestActiveSpan];
+        if (latestSpan != nil)
+            return latestSpan;
+    }
+    return self.currentTransaction;
```

Comment:
All of this needs to be synchronzied.

Remember there's a single Hub/Scope bound to the static `SentrySDK` entrypoint and all threads in the app are using it